src/features/session_replay/aggregate/index.js

```diff
@@ -0,0 +1,292 @@
+    try {
+      payload.body = gzipper(u8(stringify(payload.body)))
+      this.scheduler.opts.gzip = true
+    } catch (err) {
+      // failed to gzip
+      this.scheduler.opts.gzip = false
+    }
+    console.log('payload!', payload)
+    // TODO -- Gracefully handle the buffer for retries.
+    this.clearBuffer()
+    return [payload]
+  }
+
+  getPayload () {
+    const agentRuntime = getRuntime(this.agentIdentifier)
+    const info = getInfo(this.agentIdentifier)
+    return {
+      qs: { protocol_version: '0' },
+      body: {
+        type: 'SessionReplay',
+        appId: Number(info.applicationID),
+        timestamp: Date.now(),
+        blob: JSON.stringify(this.events), // this needs to be a stringified JSON array of rrweb nodes
+        attributes: {
+          session: agentRuntime.session.state.value,
+          hasSnapshot: this.hasSnapshot,
+          hasError: this.hasError,
+          agentVersion: agentRuntime.version,
+          isFirstChunk: this.isFirstChunk,
+          'nr.rrweb.version': RRWEB_VERSION
+        }
+      }
+    }
+  }
+
+  onHarvestFinished (result) {
+    // The mutual decision for now is to stop recording and clear buffers if ingest is experiencing 429 rate limiting
+    if (result.status === 429) {
+      this.abort()
+      return
+    }
+
+    if (this.blocked) this.scheduler.stopTimer(true)
+  }
+
+  /** Clears the buffer (this.events), and resets all payload metadata properties */
+  clearBuffer () {
+    this.events = []
+    this.isFirstChunk = false
+    this.hasSnapshot = false
+    this.hasError = false
+    this.payloadBytesEstimation = 0
+  }
+
+  /** Begin recording using configured recording lib */
+  startRecording () {
+    if (!recorder) {
+      warn('Recording library was never imported')
+      return this.abort()
+    }
+    const { blockClass, ignoreClass, maskTextClass, blockSelector, maskInputOptions, maskTextSelector, maskAllInputs } = getConfigurationValue(this.agentIdentifier, 'session_replay')
+    this.hasSnapshot = true
+    // set up rrweb configurations for maximum privacy --
+    // https://newrelic.atlassian.net/wiki/spaces/O11Y/pages/2792293280/2023+02+28+Browser+-+Session+Replay#Configuration-options
+    this.stopRecording = recorder({
+      emit: this.store.bind(this),
+      blockClass,
+      ignoreClass,
+      maskTextClass,
+      blockSelector,
+      maskInputOptions,
+      maskTextSelector,
+      maskAllInputs,
+      ...(this.mode === MODE.ERROR && { checkoutEveryNms: 30000 })
+    })
+  }
+
+  /** Store a payload in the buffer (this.events).  This should be the callback to the recording lib noticing a mutation */
+  store (event, isCheckout) {
+    if (this.blocked) return
+    const eventBytes = stringify(event).length
+    /** The estimated size of the payload after compression */
+    const payloadSize = this.getPayloadSize(eventBytes)
+    // Vortex will block payloads at a certain size, we might as well not send.
```

Comment:
Can we split the payload up and send it in multiple chunks?

Reply:
up to a certain point.  The most likely case of exceeding the limit is that the "initial snapshot", which is a single rrweb event, exceeds the limit.  Splitting that object further would be much more complex